routes/users.js

```diff
@@ -1,10 +1,38 @@
 'use strict';
 
 const express = require('express');
+const knex = require('../knex');
+const bcrypt = require('bcrypt');
 
 // eslint-disable-next-line new-cap
 const router = express.Router();
 
 // YOUR CODE HERE
 
+router.post('/users', (req, res, next) => {
+  const givenPassword = req.body.password;
+
+  bcrypt.hash(givenPassword, 12)
+    .then((result) => {
+      return knex('users').insert({
```

Comment:
before inserting a new user, check the database to check to see if that email has already been registered. If not, then register.